### ISSUE TITLE
ValidationException: Improperly formed request.

### ISSUE THREAD

**Issue by charlesivie** [aws/aws-toolkit-jetbrains#4040]
**Describe the bug**
Happens intermittently when using CodeWhisperer and Q in WebStorm

**To reproduce**

**Expected behavior**

**Screenshots**

**Your Environment**
- OS: 13.6
- JetBrains product: WebStorm
- JetBrains product version: 2023.2.5
- AWS Toolkit version: 2.0-232
- SAM CLI version:
- JVM/Python version:

**Additional context**

`software.amazon.awssdk.services.codewhispererruntime.model.ValidationException: Improperly formed request. (Service: CodeWhispererRuntime, Status Code: 400, Request ID: df8af63e-1453-4c54-94c7-1a384b254508)
	at software.amazon.awssdk.core.internal.http.CombinedResponseHandler.handleErrorResponse(CombinedResponseHandler.java:125)
	at software.amazon.awssdk.core.internal.http.CombinedResponseHandler.handleResponse(CombinedResponseHandler.java:82)
	at software.amazon.awssdk.core.internal.http.CombinedResponseHandler.handle(CombinedResponseHandler.java:60)
	at software.amazon.awssdk.core.internal.http.CombinedResponseHandler.handle(CombinedResponseHandler.java:41)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.HandleResponseStage.execute(HandleResponseStage.java:40)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.HandleResponseStage.execute(HandleResponseStage.java:30)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallAttemptTimeoutTrackingStage.execute(ApiCallAttemptTimeoutTrackingStage.java:72)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallAttemptTimeoutTrackingStage.execute(ApiCallAttemptTimeoutTrackingStage.java:42)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.TimeoutExceptionHandlingStage.execute(TimeoutExceptionHandlingStage.java:78)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.TimeoutExceptionHandlingStage.execute(TimeoutExceptionHandlingStage.java:40)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallAttemptMetricCollectionStage.execute(ApiCallAttemptMetricCollectionStage.java:52)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallAttemptMetricCollectionStage.execute(ApiCallAttemptMetricCollectionStage.java:37)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.RetryableStage.execute(RetryableStage.java:81)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.RetryableStage.execute(RetryableStage.java:36)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.StreamManagingStage.execute(StreamManagingStage.java:56)
	at software.amazon.awssdk.core.internal.http.StreamManagingStage.execute(StreamManagingStage.java:36)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallTimeoutTrackingStage.executeWithTimer(ApiCallTimeoutTrackingStage.java:80)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallTimeoutTrackingStage.execute(ApiCallTimeoutTrackingStage.java:60)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallTimeoutTrackingStage.execute(ApiCallTimeoutTrackingStage.java:42)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallMetricCollectionStage.execute(ApiCallMetricCollectionStage.java:48)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallMetricCollectionStage.execute(ApiCallMetricCollectionStage.java:31)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ExecutionFailureExceptionReportingStage.execute(ExecutionFailureExceptionReportingStage.java:37)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ExecutionFailureExceptionReportingStage.execute(ExecutionFailureExceptionReportingStage.java:26)
	at software.amazon.awssdk.core.internal.http.AmazonSyncHttpClient$RequestExecutionBuilderImpl.execute(AmazonSyncHttpClient.java:196)
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.invoke(BaseSyncClientHandler.java:103)
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.doExecute(BaseSyncClientHandler.java:171)
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.lambda$execute$1(BaseSyncClientHandler.java:82)
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.measureApiCallSuccess(BaseSyncClientHandler.java:179)
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.execute(BaseSyncClientHandler.java:76)
	at software.amazon.awssdk.core.client.handler.SdkSyncClientHandler.execute(SdkSyncClientHandler.java:45)
	at software.amazon.awssdk.awscore.client.handler.AwsSyncClientHandler.execute(AwsSyncClientHandler.java:56)
	at software.amazon.awssdk.services.codewhispererruntime.DefaultCodeWhispererRuntimeClient.sendTelemetryEvent(DefaultCodeWhispererRuntimeClient.java:612)
	at software.amazon.awssdk.services.codewhispererruntime.CodeWhispererRuntimeClient.sendTelemetryEvent(CodeWhispererRuntimeClient.java:1185)
	at software.aws.toolkits.jetbrains.services.codewhisperer.credentials.CodeWhispererClientAdaptorImpl.sendMetricDataTelemetry(CodeWhispererClientAdaptor.kt:310)
	at software.aws.toolkits.jetbrains.services.codewhisperer.telemetry.CodeWhispererUserModificationTracker.sendModificationWithChatTelemetry(CodeWhispererUserModificationTracker.kt:208)
	at software.aws.toolkits.jetbrains.services.codewhisperer.telemetry.CodeWhispererUserModificationTracker.emitTelemetryOnChatCodeInsert(CodeWhispererUserModificationTracker.kt:128)
	at software.aws.toolkits.jetbrains.services.codewhisperer.telemetry.CodeWhispererUserModificationTracker.flush(CodeWhispererUserModificationTracker.kt:99)
	at software.aws.toolkits.jetbrains.services.codewhisperer.telemetry.CodeWhispererUserModificationTracker.scheduleCodeWhispererTracker$lambda$0(CodeWhispererUserModificationTracker.kt:69)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:254)
	at com.intellij.util.Alarm$Request.runSafely(Alarm.java:373)
	at com.intellij.util.Alarm$Request.run(Alarm.java:360)
	at com.intellij.util.concurrency.Propagation.contextAwareCallable$lambda$2(propagation.kt:328)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.util.concurrency.SchedulingWrapper$MyScheduledFutureTask.run(SchedulingWrapper.java:272)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:249)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:31)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.executeFirstTaskAndHelpQueue(BoundedTaskExecutor.java:227)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:218)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:215)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:840)
`

**Comment by virtualgill** [comment]
I'm seeing this too since I enabled Q.
OS: macOS Sonoma v14.2
JetBrains product: IntelliJ
JetBrains product version: 2023.1.5 (Ultimate Edition)
AWS Toolkit version: 2.1-231

**Comment by 094459** [comment]
I am also seeing it in my setup:
OS: macos Ventura 13.6.4
Jetbrains product: IntelliJ
Jetbrains product version: 2023.3.5 (Ultimate Edition)
AWS Toolkit version -2.13-233

**Comment by andrewyuq** [comment]
@094459 could you grab debug logs in your IDE?
https://builderhub.corp.amazon.com/docs/codewhisperer/user-guide/troubleshooting.html

**Comment by Will-ShaoHua** [comment]
repro step
1. open up a file which Q doesn't recognize, here i'm using `gradle.properties`
2. turn on auto-scan (previously it was disabled)
3. auto scan starts running and error pops up


```
2024-06-11 14:59:44,838 [9240663]   FINE - software.aws.toolkits.jetbrains.services.codewhisperer.codescan.CodeWhispererCodeScanManager - Language is unknown or plaintext, skipping code scan.
2024-06-11 14:59:44,839 [9240664]   FINE - software.aws.toolkits.jetbrains.services.codewhisperer.telemetry.CodeWhispererTelemetryService - Recording code security scan event. 
Total number of security scan issues found: 0, 
Number of security scan issues with fixes: 0, 
Language: unknown, 
Uncompressed source payload size in bytes: 0, 
Uncompressed build payload size in bytes: null, 
Compressed source zip file size in bytes: 0, 
Total project size in bytes: 0.0, 
Total duration of the security scan job in milliseconds: 2.0, 
Context truncation duration in milliseconds: 0, 
Artifacts upload duration in milliseconds: 0, 
Service invocation duration in milliseconds: 0, 
Total number of lines scanned: 0, 
Reason: null 
Scope: FILE 
Passive: true 
 
2024-06-11 14:59:45,328 [9241153]   FINE - software.aws.toolkits.jetbrains.services.codewhisperer.codescan.CodeWhispererCodeScanManager - Failed to send code scan telemetry. RequestId: a57d1fea-5e6d-4b10-bdb3-92adbc3697e5, ErrorMessage: Improperly formed request. (Service: CodeWhispererRuntime, Status Code: 400, Request ID: a57d1fea-5e6d-4b10-bdb3-92adbc3697e5)
```

**Comment by Will-ShaoHua** [comment]
Fix has been released with amazonQ plugin version 3.11
